pyfr/writers/native.py

```diff
@@ -36,7 +37,8 @@ def __init__(self, intg, basedir, basename, prefix, *, extn='.pyfrs'):
             self.basename += extn
 
         # Output counter (incremented each time write() is called)
-        self.nout = self._restore_nout() if intg.isrestart else 0
+        self.fgen = file_path_gen(self.basedir, self.basename)
```

Comment:
I think we need to preserve the old functionality.  Maybe add a `restore=True|False` argument to the generator?

Reply:
Also, no need to persist basedir/basename since the generator now takes care of that state for us.